### PR TITLE
Component IDs

### PR DESCRIPTION
**Pull request by GeorgeHahn**
### What does this PR do?

Add optional component IDs to the top-level generator & blackhole config.

### Motivation

This will allow us to differentiate between components.

### Related issues

SMP-575

### CODE REVIEW

File: src/blackhole/http.rs [hunk 1]
```diff
@@ -114,11 +104,12 @@ struct KinesisPutRecordBatchResponse {
 #[allow(clippy::borrow_interior_mutable_const)]
 async fn srv(
     status: StatusCode,
+    labels: Arc<Vec<(String, String)>>,
     body_variant: BodyVariant,
     req: Request<Body>,
     headers: HeaderMap,
 ) -> Result<Response<Body>, hyper::Error> {
-    requests_received().increment(1);
+    counter!("bytes_received", 1, &*labels);
```
Comment (blt) [inline src/blackhole/http.rs:112]: Hmm, I hate to go back to `counter!`. It's actually not terribly cheap to create these things on the fly like this. At least I managed to drop lading's CPU use by ~30% when generating dogstatsd metrics over UDS.

Comment (blt) [reply src/blackhole/http.rs:112]: I wonder, could we maybe avoid passing `labels` and instead pass the metrics handle itself, pre-loaded with labels?

**Review by blt — COMMENTED** [review]
On the whole this looks excellent. I do regret the need to go back to the use of `counter!` and `gauge!` and wonder if we couldn't contrive to avoid them.

File: src/blackhole/http.rs (hunk 1 above)
Comment (GeorgeHahn) [reply src/blackhole/http.rs]: Good call, that worked nicely.

**Review by blt — APPROVED** [review]

**Pull request merged**